### DISCUSSION TITLE
File executed with cx-frezze showing error "library not loaded"

### DISCUSSION THREAD

**Discussion by pennyxxp** [marcelotduarte/cx_Freeze#1053] [Q&A]
Hello, I am very new to programming. 
My application file executed with cx-frezze shows error "library not loaded". Are there anything that I can do with my code or my setting?
Thank you very much :'(

I followed a tutorial and wrote a simple game with python using pygame on pycharm, the program runs well on pycharm. Yet the application file executed with cx-freeze shows error "library not loaded" ,"Reason: no suitable image found".

It seems that there were people talking about these errors long time ago (I googled these error phrases) yet I still couldn't find ways to fix them. 
(mac os10.15.7, cx-freeze6.6, python3.9, pygame 2.0.1)

# error message
Dyld Error Message:
  Library not loaded: @executable_path/lib/Python
  Referenced from: /Users/USER/*/KUSO GAME-0.1.app/Contents/MacOS/main
  Reason: no suitable image found.  Did find:

/Users/penny/PycharmProjects/pygame2/build/KUSO GAME-0.1.app/Contents/MacOS/lib/Python: code signature invalid for '/Users/penny/PycharmProjects/pygame2/build/KUSO GAME-0.1.app/Contents/MacOS/lib/Python'

/Users/penny/PycharmProjects/pygame2/build/KUSO GAME-0.1.app/Contents/MacOS/lib/Python: code signature invalid for '/Users/penny/PycharmProjects/pygame2/build/KUSO GAME-0.1.app/Contents/MacOS/lib/Python'

/Users/penny/PycharmProjects/pygame2/build/KUSO GAME-0.1.app/Contents/MacOS/lib/Python: stat() failed with errno=1

# setup.py
application_title = "KUSO GAME"
main_python_file = "main.py"
import sys
from cx_Freeze import setup, Executable

base = None
if sys.platform == "win32":
    base = "Win32GUI"

includes = ["atexit","re"]

setup(
    name = application_title,
    version = "0.1",
    description = "",
    options = {"build_exe": {"includes": includes}},
    executables = [Executable(main_python_file, base=base)])

**Comment by marcelotduarte** [comment]
In #809 the developer is using pygame, and have sucess in cx-freeze 6.4.2
If you're building the project with:
python setup.py build
Try to use
python setup.py bdist_mac
or
python setup.py bdist_dmg

**Comment by luziferius** [comment]
> Did find:
>  code signature invalid

Is your Mac configured to only allow execution of signed and notarized applications?
There is a recent trend towards only allowing execution of signed (and notarized) applications. As far as I heard, the default macOS security setting was changed in a recent update to only allow notarized applications.

The error message indicates that it cant find a suitable image, because the found images are not properly signed.

Try configuring your device to allow execution of unsigned applications and then try again..
